src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementTileEntityUpdater.java

```diff
@@ -105,62 +92,34 @@ public void startTimer() {
   }
 
   private void startTimer(int duration) {
-    stopped = false;
     TimerConfig timerConfig = new TimerConfig();
     timerConfig.setPersistent(false);
     timerService.createSingleActionTimer(duration, timerConfig);
   }
 
-  @Override
-  public void stopTimer() {
-    stopped = true;
-  }
-  
-  @Asynchronous
-  public void onTileIdUpdateRequest(@Observes TileIdUpdateRequest event) {
-    if (!stopped) {
-      OrganizationId organizationId = event.getOrganizationId();
-      
-      if (organizationSettingController.getSettingValue(organizationId, ManagementConsts.ORGANIZATION_SETTING_BASEURL) == null) {
-        return;
-      }
-      
-      queue.add(event);
+  @Timeout
+  public void timeout(Timer timer) {
+    if (systemSettingController.isNotTestingOrTestRunning()) {
+      executeNextTask();
     }
+
+    startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
   }
   
-  @Asynchronous
-  public void onTileIdRemoveRequest(@Observes TileIdRemoveRequest event) {
-    if (!stopped) {
-      TileId tileId = event.getId();
-      
-      if (!StringUtils.equals(tileId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      deleteTile(event.getOrganizationId(), tileId);
-    }
-  }
-
-  @Timeout
-  public void timeout(Timer timer) {
-    if (!stopped) {
-      if (systemSettingController.isNotTestingOrTestRunning()) {
-        TileIdUpdateRequest updateRequest = queue.next();
-        if (updateRequest != null) {
-          updateManagementTile(updateRequest);
-        }
+  private void executeNextTask() {
+    IdTask<TileId> task = tileIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateManagementTile(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteManagementTile(task.getId());
       }
-
-      startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
 
-  private void updateManagementTile(TileIdUpdateRequest updateRequest) {
-    OrganizationId organizationId = updateRequest.getOrganizationId();
+  private void updateManagementTile(TileId managementTileId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateManagementTile" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)